growth/too/tasks/growthdb_cgi.py

```diff
@@ -47,77 +140,257 @@ def get_source_autoannotations(sourceid):
+            CLU_sfr_ha = None
+        try:
+            CLU_sfr_fuv = float(s['autoannotations_dict']['CLU_sfr_fuv'])
+        except (TypeError, KeyError, ValueError):
+            CLU_sfr_fuv = None
+        try:
+            CLU_z = float(s['autoannotations_dict']['CLU_z'])
+        except (TypeError, KeyError, ValueError):
+            CLU_z = None
+        try:
+            CLU_mstar = float(s['autoannotations_dict']['CLU_mstar'])
+        except (TypeError, KeyError, ValueError):
+            CLU_mstar = None
+        try:
+            CLU_d_to_galaxy_arcsec = float(
+                s['autoannotations_dict']['CLU_d_to_galaxy_arcsec'])
+        except (TypeError, KeyError, ValueError):
+            CLU_d_to_galaxy_arcsec = None
+        try:
+            CLU_name = str(s['autoannotations_dict']['CLU_name'])
+        except (TypeError, KeyError, ValueError):
+            CLU_name = None
+        try:
+            CLU_id = int(s['autoannotations_dict']['CLU_id'])
+        except (TypeError, KeyError, ValueError):
+            CLU_id = None
 
-        models.db.session.merge(
-            models.Candidate(
-                name=s['name'],
-                subfield_id=rcid,
-                creationdate=creationdate,
-                classification=s['classification'],
-                redshift=redshift,
-                iauname=s['iauname'],
-                field_id=field,
-                candid=candid,
-                ra=float(s['ra']),
-                dec=float(s['dec']),
-                lastmodified=lastmodified,
-                autoannotations=s['autoannotations']
-                )
-        )
+        kwargs = {'name': s['name'],
+                  'subfield_id': rcid,
+                  'creationdate': creationdate,
+                  'classification': s['classification'],
+                  'redshift': redshift,
+                  'iauname': s['iauname'],
+                  'field_id': field,
+                  'candid': candid,
+                  'ra': float(s['ra']),
+                  'dec': float(s['dec']),
+                  'lastmodified': lastmodified,
+                  'autoannotations': s['autoannotations'],
+                  'ps1_dr2_detections': ps1_dr2_detections,
+                  'gaia_match': gaia_match,
+                  'gaia_parallax': gaia_parallax,
+                  'w4': w4,
+                  'w2mw3': w2mw3,
+                  'w1mw2': w1mw2,
+                  'jdstarthist': jdstarthist,
+                  'ssmagnr': ssmagnr,
+                  'ssdistnr': ssdistnr,
+                  'CLU_sfr_ha': CLU_sfr_ha,
+                  'CLU_mstar': CLU_mstar,
+                  'CLU_sfr_fuv': CLU_sfr_fuv,
+                  'CLU_z': CLU_z,
+                  'CLU_name': CLU_name,
+                  'CLU_id': CLU_id,
+                  'CLU_d_to_galaxy_arcsec': CLU_d_to_galaxy_arcsec
+                  }
+        candidate = models.Candidate.query.filter_by(name=s['name']).all()
```

Comment:
This query is inside a `for:` loop over the candidates, so it is going to be O(N^2) in the number of candidates. This is not scalable.

Reply:
Structure changed, now it should be scalable.